code/modules/mob/living/simple_animal/hostile/roguetown/jitterskull.dm

```diff
@@ -0,0 +1,1349 @@
+// Jitterskull — a small, jittery floating skull for Halloween
+
+/mob/living/simple_animal/hostile/rogue/jitterskull
+    name = "jitterskull"
+    desc = "A ghoul of some kind, probably from a forest."
+    icon = 'icons/roguetown/mob/monster/jitterskull.dmi'
+    icon_state = "jitany"
+    icon_living = "jitany"
+    icon_dead = null
+    mob_biotypes = MOB_UNDEAD|MOB_SPIRIT
+    movement_type = FLYING
+    environment_smash = ENVIRONMENT_SMASH_NONE
+    pass_flags = PASSTABLE|PASSGRILLE
+    // Use a blunt, non-cutting base intent; custom bite damage is applied separately
+    // Jitterskull uses a special crush intent that rips through armor and heavily damages integrity
+    base_intents = list(/datum/intent/simple/jitterskull_crush)
+    gender = NEUTER
+    speak_chance = 0
+    // Move faster to help it secure hits
+    turns_per_move = 2
+    response_help_continuous = "passes through"
+    response_help_simple = "pass through"
+    maxHealth = 1000
+    health = 1000
+    spacewalk = TRUE
+    stat_attack = UNCONSCIOUS
+    robust_searching = 1
+    // Faster than average
+    speed = -1
+    move_to_delay = 1
+    harm_intent_damage = 1
+    obj_damage = 1
+    // Use custom bite attack; keep minimal base damage so base attack flow (hit/miss/defense) still runs
+    // We'll apply the heavy damage ourselves in AttackingTarget()
+    melee_damage_lower = 25
+    melee_damage_upper = 90
+    d_type = "blunt"
+    attack_verb_simple = "bites"
+    attack_verb_continuous = "bites"
+    armor_penetration = 100
+    // It attacks anything and everything
+    attack_same = TRUE
+    attack_sound = 'sound/combat/wooshes/bladed/wooshmed (1).ogg'
+    dodge_sound = 'sound/combat/dodge.ogg'
+    parry_sound = "bladedmedium"
+    d_intent = INTENT_PARRY
+    speak_emote = list("chatters")
+    del_on_death = TRUE
+    STALUC = 10
+    STASTR = 20
+    STACON = 20
+    STASPD = 18
+    atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+    minbodytemp = 0
+    // Empty faction so basic targeting treats everything as valid
+    faction = list()
+    footstep_type = null
+    defprob = 40
+    canparry = TRUE
+    retreat_health = null
+    can_have_ai = FALSE // disable native simple_animal AI; we use an ai_controller
+    AIStatus = AI_OFF
+    ai_controller = /datum/ai_controller/haunt
+
+    // Behavior control
+    var/next_attack_time = 0       // world.time until which we will refrain from attacking
+    var/teleport_cooldown_until = 0 // throttle for on-hit teleport reactions
+    var/next_idle_chatter_time = 0  // next time we're allowed to idle-chatter
+    var/taunt_cooldown_until = 0    // cooldown for taunting after a down
+    var/is_dying = FALSE            // guard to prevent re-entrant death sequences
+    var/mob/living/reengage_target  // who we plan to reengage after a panic blink
```

Comment:
potential harddel source, this should 1) be handled on an AI controller/AI behavior datum, and 2) use a weakref to avoid harddels